source/d3d9/buffer_detection.hpp

```diff
@@ -39,12 +43,12 @@ namespace reshade::d3d9
 		void on_set_depthstencil(IDirect3DSurface9 *&depthstencil);
 		void on_get_depthstencil(IDirect3DSurface9 *&depthstencil);
 		void on_clear_depthstencil(UINT clear_flags);
+		void on_set_viewport(D3DVIEWPORT9 viewport);
 
 		// Detection Settings
 		bool disable_intz = false;
 		bool preserve_depth_buffers = false;
-		UINT depthstencil_clear_index = std::numeric_limits<UINT>::max();
-		UINT depthstencil_clear_index_override = 0;
+		std::pair<com_ptr<IDirect3DSurface9>, UINT> depthstencil_clear_index = { nullptr, 0 };
```

Comment:
I don't think this needs to be a `com_ptr`. There will always be a reference held by `_depthstencil_replacement` or `_preserved_depthstencil_surfaces`. And not having this a `com_ptr` avoid having to release the reference in `buffer_detection::reset`.

Reply:
OK. Done.